File: frontend/src/types/competitor.ts
export type UserSentiment = 'Positive' | 'Mixed' | 'Negative';

export interface Competitor {
  id: string;
  created_time: string;
  competitor_name: string;
  website: string;
  key_features: string;
  pricing: string;
  user_sentiment: UserSentiment;
}

File: backend/models/competitor.py
from typing import Literal

from pydantic import BaseModel


UserSentiment = Literal["Positive", "Mixed", "Negative"]


class CompetitorCreate(BaseModel):
    competitor_name: str
    website: str
    key_features: str
    pricing: str
    user_sentiment: UserSentiment


class Competitor(CompetitorCreate):
    id: str
    created_time: str
